Code/Framework/AzToolsFramework/AzToolsFramework/Metadata/MetadataManager.h

```diff
@@ -0,0 +1,101 @@
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace AzToolsFramework
+{
+    struct IMetadataRequests
+    {
+        AZ_RTTI(IMetadataRequests, "{88EB4C71-F2A5-48C6-A08B-0D6A8EB08A74}");
+
+        virtual ~IMetadataRequests() = default;
+
+        //! Gets a value from the metadata file associated with the file.
+        //! @param file Absolute path to the file (or metadata file).
+        //! @param key JSONPath formatted key for the metadata value to read.  Ex: /Settings/Platform/pc.
+        //! @param typeId Type of the stored value and outValue.
+        //! @return True if metadata file and key exists and was successfully read, false otherwise.
+        virtual bool GetValue(AZ::IO::PathView file, AZStd::string_view key, void* outValue, AZ::Uuid typeId) = 0;
+
+        //! Gets a value from the metadata file associated with the file.
+        //! @param file Absolute path to the file (or metadata file).
+        //! @param key JSONPath formatted key for the metadata value to read.  Ex: /Settings/Platform/pc.
+        //! @return True if metadata file and key exists and was successfully read, false otherwise.
+        template<typename T>
+        bool GetValue(AZ::IO::PathView file, AZStd::string_view key, T& outValue)
+        {
+            return GetValue(file, key, &outValue, azrtti_typeid<T>());
+        }
+
+        //! Gets a JSON Value from the metadata file associated with the file.
+        //! Prefer to use GetValue instead where possible.  This is best used for reading old versions.
+        //! @param file Absolute path to the file (or metadata file).
+        //! @param key JSONPath formatted key for the metadata value to read.  Ex: /Settings/Platform/pc.
+        //! @return True if metadata file and key exists and was successfully read, false otherwise.
+        virtual bool GetJsonValue(AZ::IO::PathView file, AZStd::string_view key, rapidjson_ly::Value& outValue) = 0;
+
+        //! Gets the version for a stored key/value from the metadata file associated with the file.
+        //! @param file Absolute path to the file (or metadata file).
+        //! @param key JSONPath formatted key for the metadata value to read.  Ex: /Settings/Platform/pc.
+        //! @return True if metadata file and key exists and was successfully read, false otherwise.
+        virtual bool GetValueVersion(AZ::IO::PathView file, AZStd::string_view key, int& version) = 0;
+
+        //! Sets a value in the metadata file associated with the file.
+        //! @param file Absolute path to the file (or metadata file).
+        //! @param key JSONPath formatted key for the metadata value to write.  Ex: /Settings/Platform/pc.
+        //! @param typeId Type of the stored value and inValue.
+        //! @return True if the value is successfully saved to disk, false otherwise.
+        virtual bool SetValue(AZ::IO::PathView file, AZStd::string_view key, const void* inValue, AZ::Uuid typeId) = 0;
```

Comment:
I'm wondering whether the API should allow to set the value version as well. If all the key/value pairs share the same version, why don't we just keep the version of the metadata file itself?

Reply:
Discussed offline - There was confusion between the 2 versions, I've updated the comments to try and make it more clear.  The metadata file version is appended to the document as its own key.  Each key/value pair has its type's version appended to it to allow some rough version conversion.  The version isn't per field, so an update to a "type within a type" will not show a different version.

Reply:
I read this "GetValueVersion()" and got the idea that the version was attached to the "Value" but if the value is Bool or Float how would that work? If the Value is a TypeT, then that sort of makes sense.

Reply:
The version is only added for types which serialize as Json Objects.  So simple types like bool wouldn't have a version attached